### PR TITLE
[Warnings] Removing warnings appearing in `feature_fetcher` tests.

### PR DESCRIPTION
**Pull request by drivanov**
## Description

The following warnings:
```
tests/python/pytorch/graphbolt/test_feature_fetcher.py::test_FeatureFetcher_with_edges_homo
tests/python/pytorch/graphbolt/test_feature_fetcher.py::test_FeatureFetcher_with_edges_hetero
  /usr/local/lib/python3.10/dist-packages/torch/utils/data/datapipes/utils/common.py:144: UserWarning: Local function 
is not supported by pickle, please use regular python function or functools.partial instead.
    warnings.warn(
```
which appeared in two `FeatureFetcher` tests, have been fixed.
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by frozenbugs** [comment]
@dgl-bot

**Comment by frozenbugs** [comment]
```
=================================== ERRORS ====================================
___ ERROR collecting tests/python/pytorch/graphbolt/test_feature_fetcher.py ___
tests\python\pytorch\graphbolt\test_feature_fetcher.py:6: in &lt;module&gt;
    from torchdata.datapipes.iter import Mapper
C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\site-packages\torchdata\__init__.py:9: in &lt;module&gt;
    from . import datapipes
C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\site-packages\torchdata\datapipes\__init__.py:9: in &lt;module&gt;
    from . import iter, map, utils
C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\site-packages\torchdata\datapipes\iter\__init__.py:99: in &lt;module&gt;
    from torchdata.datapipes.iter.util.distributed import FullSyncIterDataPipe as FullSync
C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\site-packages\torchdata\datapipes\iter\util\distributed.py:23: in &lt;module&gt;
    from torchdata.datapipes.iter.util.prefetcher import PRODUCER_SLEEP_INTERVAL
C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\site-packages\torchdata\datapipes\iter\util\prefetcher.py:33: in &lt;module&gt;
    @functional_datapipe("prefetch")
C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\site-packages\torch\utils\data\datapipes\_decorator.py:36: in __call__
    IterDataPipe.register_datapipe_as_function(self.name, cls, enable_df_api_tracing=self.enable_df_api_tracing)
C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\site-packages\torch\utils\data\datapipes\datapipe.py:135: in register_datapipe_as_function
    raise Exception(f"Unable to add DataPipe function name {function_name} as it is already taken")
E   Exception: Unable to add DataPipe function name prefetch as it is already taken
============================== warnings summary ===============================
python\dgl\backend\backend.py:1838
  C:\Jenkins\workspace\dgl_PR-7633\python\dgl\backend\backend.py:1838: DeprecationWarning: invalid escape sequence '\P'
    """Segment reduction operator.

..\..\..\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\site-packages\outdated\__init__.py:36
  C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\site-packages\outdated\__init__.py:36: DeprecationWarning: pkg_resources is deprecated as an API. See <a href='https://setuptools.pypa.io/en/latest/pkg_resources.html'>https://setuptools.pypa.io/en/latest/pkg_resources.html</a>
    from pkg_resources import parse_version

..\..\..\Users\Administrator\Envs\jenkins-dgl-PR-7633-4\Lib\site-packages\pkg_resources\__init__.py:3117
  C:\Users\Administrator\Envs\jenkins-dgl-PR-7633-4\Lib\site-packages\pkg_resources\__init__.py:3117: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('sphinxcontrib')`.
  Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See <a href='https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages'>https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages</a>
    declare_namespace(pkg)

-- Docs: <a href='https://docs.pytest.org/en/stable/how-to/capture-warnings.html'>https://docs.pytest.org/en/stable/how-to/capture-warnings.html</a>
--- generated xml file: C:\Jenkins\workspace\dgl_PR-7633\pytest_backend.xml ---
=========================== short test summary info ===========================
ERROR tests/python/pytorch/graphbolt/test_feature_fetcher.py - Exception: Unable to add DataPipe function name prefetch as it is already taken
!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
======================== 3 warnings, 1 error in 5.25s =========================
```

File: tests/python/pytorch/graphbolt/test_feature_fetcher.py [hunk 1]
```diff
@@ -1,10 +1,9 @@
 import random
-from enum import Enum
+from functools import partial
 
 import dgl.graphbolt as gb
-import pytest
 import torch
-from torch.utils.data.datapipes.iter import Mapper
+from torchdata.datapipes.iter import Mapper
```
Comment (mfbalin) [inline tests/python/pytorch/graphbolt/test_feature_fetcher.py:6]:
```suggestion
from torch.utils.data.datapipes.iter import Mapper
```

Comment (drivanov) [reply tests/python/pytorch/graphbolt/test_feature_fetcher.py]: Done

**Comment by mfbalin** [comment]
@dgl-bot

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**